msal/application.py

```diff
@@ -231,8 +231,23 @@ def __init__(
 
         :param str authority:
             A URL that identifies a token authority. It should be of the format
-            https://login.microsoftonline.com/your_tenant
-            By default, we will use https://login.microsoftonline.com/common
+            ``https://login.microsoftonline.com/your_tenant``
+            By default, we will use ``https://login.microsoftonline.com/common``
```

Comment:
>By default, we will use ``https://login.microsoftonline.com/common`` [](http://example.com/codeflow?start=12&length=68)

for CCA it should be tenanted and common should not be allowed?

Reply:
It is true that AAD recommend to avoid "common" for confidential client, yet at the same time they can't really block such a usage due to backward compatibility concern.

We are essentially in the same boat here. Historically the input parameter `authority` is optional and defaults to that ".../common" value. We have, however, long been updated our samples to guide app developers to use their tenant. See [here](https://github.com/AzureAD/microsoft-authentication-library-for-python/blob/1.16.0/sample/confidential_client_secret_sample.py#L5) and [there](https://github.com/AzureAD/microsoft-authentication-library-for-python/blob/1.16.0/sample/confidential_client_certificate_sample.py#L5). In practice, few people would read our API Reference doc (sad but true :-) ) but they will typically start from our samples. So, I think we are good.